src/main/scala/com/fulcrumgenomics/umi/UmiConsensusCaller.scala

```diff
@@ -216,7 +216,14 @@ trait UmiConsensusCaller[C <: SimpleRead] {
     // Find the last non-N base of sufficient quality in the record, starting from either the
     // end of the read, or the end of the insert, whichever is shorter!
     val len = {
-      var index = if (rec.isFrPair) min(abs(rec.insertSize), trimToLength) - 1 else trimToLength - 1
+      // Adjust the insert size based on clipping on the 5' end of the given read.  This *does not* consider any clipping
+      // on the 5' end of its mate.
+      val adjustedInsertSize = if (!rec.isFrPair) bases.length else {
```

Comment:
I don't love how this has a second check for `rec.isFrPair`, and then passes back the read length.  How about this instead for calculating `index`:

```scala
var index = if (!rec.isFrPair) trimToLength - 1 else {
  // Adjust the insert size based on clipping on the 5' end of the given read.
  // This *does not* consider any clipping on the 5' end of its mate.
  val fivePrimeClipping  = cigar.takeWhile(_.operator.isClipping).filter(_.operator == CigarOperator.S).sumBy(_.length)
  val adjustedInsertSize = abs(rec.insertSize) + fivePrimeClipping
  min(adjustedInsertSize, trimToLength) - 1
}
```

Reply:
Also, note that this removes the `val cigar = ...` because that exact thing is already don on lines 199-203.